sw/device/tests/dif/dif_otbn_unittest.cc

```diff
@@ -332,32 +332,32 @@ TEST_F(GetErrCodeTest, NullArgs) {
   result = dif_otbn_get_err_code(&dif_otbn_, nullptr);
   EXPECT_EQ(result, kDifOtbnBadArg);
 
-  err_code = kDifOtbnErrorCodeNoError;
+  err_code = kDifOtbnErrCodeBadDataAddr;
   result = dif_otbn_get_err_code(nullptr, &err_code);
   EXPECT_EQ(result, kDifOtbnBadArg);
```

Comment:
This isn't changed in this PR, but you should not write
```cpp
auto result = function_under_test(...);
EXPECT_THAT(result, ...);
```
since the EXPECT macros can give better error messages when `result` is inlined.

Reply:
This pattern is quite common in this file, so if we want to change something along those lines, we should do so in a separate PR.

That being said, I don't see how inlining the function call makes the error reporting significantly better. To illustrate that, here's an example.

```c
result = dif_otbn_get_err_code(nullptr, &err_code);
EXPECT_EQ(result, kDifOtbnDisable);
EXPECT_EQ(err_code, kDifOtbnErrCodeBadDataAddr);
```
```
../sw/device/tests/dif/dif_otbn_unittest.cc:337: Failure
Expected equality of these values:
  result
    Which is: 2
  kDifOtbnDisable
    Which is: 1
```

vs.

```c
EXPECT_EQ(dif_otbn_get_err_code(nullptr, &err_code), kDifOtbnDisable);
EXPECT_EQ(err_code, kDifOtbnErrCodeBadDataAddr);
```
```
../sw/device/tests/dif/dif_otbn_unittest.cc:336: Failure
Expected equality of these values:
  dif_otbn_get_err_code(nullptr, &err_code)
    Which is: 2
  kDifOtbnDisable
    Which is: 1
```

* I don't find the error message significantly more helpful. It saves me from opening the file and looking at the line that is reported. But for me, that's the first step when I see an assert failure anyways.
* In cases where I need to check more than one value after the function call (output arguments), I need to have multiple statements anyways. Using an intermediate variable for the function result makes the checking more consistent IMO.

Reply:
> This pattern is quite common in this file, so if we want to change something along those lines, we should do so in a separate PR.

That's fine. However, I'm trying to get everyone to write expectations in a consistent manner, and the latter form is idomatic googletest.

> It saves me from opening the file and looking at the line that is reported.

This is exactly the reason EXPECTs should be written this way, because tests are run in many contexts, not just one where you are already looking at the test file. The more information is present in a failure in CI, the better. That this information is useful is subjective, but the cost of including it is low and I'm already refactoring tests to use this style anyway.